File: nom-api/Nom.Orch/Models/Recipe/RecipeAdvancedSearchModel.cs
using System.Collections.Generic;

namespace Nom.Orch.Models.Recipe
{
    /// <summary>
    /// Advanced search model for recipes with comprehensive filtering options
    /// </summary>
    public class RecipeAdvancedSearchModel
    {
        public string? Query { get; set; }
        public List<long>? CategoryIds { get; set; }
        public List<long>? TagIds { get; set; }
        public List<long>? ToolIds { get; set; }
        public List<long>? IngredientIds { get; set; }
        public List<long>? CuisineTypeIds { get; set; }
        public List<long>? HouseholdIds { get; set; }
        public int? MinRating { get; set; }
        public int? MaxPrepTime { get; set; }
        public int? MaxCookTime { get; set; }
        public int? MaxTotalTime { get; set; }
        public bool? IsPublic { get; set; }
        public bool? IsApproved { get; set; }
        public string? SortBy { get; set; }
        public string? SortDirection { get; set; }
        public int Page { get; set; } = 1;
        public int PageSize { get; set; } = 20;
        public bool IncludeIngredients { get; set; } = false;
        public bool IncludeSteps { get; set; } = false;
        public bool IncludeNutrition { get; set; } = false;
    }
} 

File: nom-ui/src/app/recipe/models/recipe-advanced-search.model.ts
export interface RecipeAdvancedSearchModel {
    query?: string;
    categoryIds?: number[];
    tagIds?: number[];
    toolIds?: number[];
    ingredientIds?: number[];
    cuisineTypeIds?: number[];
    householdIds?: number[];
    minRating?: number;
    maxPrepTime?: number;
    maxCookTime?: number;
    maxTotalTime?: number;
    isPublic?: boolean;
    isApproved?: boolean;
    sortBy?: string;
    sortDirection?: string;
    page: number;
    pageSize: number;
    includeIngredients: boolean;
    includeSteps: boolean;
    includeNutrition: boolean;
}

export interface RecipeSuggestionQueryModel {
    query?: string;
    foodIds?: number[];
    toolIds?: number[];
    limit: number;
    includeIngredients: boolean;
    includeSteps: boolean;
}

export interface RecipeSuggestionResultModel {
    id: number;
    name: string;
    description?: string;
    imageUrl?: string;
    rating?: number;
    ratingCount: number;
    categories: string[];
    tags: string[];
    ingredients?: RecipeSuggestionIngredient[];
    steps?: RecipeSuggestionStep[];
}

export interface RecipeSuggestionIngredient {
    id: number;
    name: string;
    quantity?: number;
    measurement?: string;
    notes?: string;
}

export interface RecipeSuggestionStep {
    id: number;
    stepNumber: number;
    instructions: string;
    imageUrl?: string;
}

export interface RecipeSuggestionResponseModel {
    suggestions: RecipeSuggestionResultModel[];
    totalCount: number;
}
